orchard/modules/_branch.py

```diff
@@ -6,6 +6,228 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+import pprint
+import yaml
+from pathlib import Path
+import os
 
-class Branch:
-    pass
+
+# Gets a named module from a link or config file
+def get_module(file_data, name):
+    return next((item for item in file_data['modules'] if item['name'] == name), None)
+
+
+# Returns the depth level of a given module based on the link file
+def depth_level(link_file_data, module):
+    current_depth = 0
+    if 'dependencies' in module:
+        for mod_name in module['dependencies']:
+            current_depth = max(current_depth, depth_level(link_file_data, get_module(link_file_data, mod_name)))
+    return current_depth + 1
+
+
+# Returns true if an argument causes branching in a link module
+def is_branching_arg(link_file_mod, argname):
+    for arg in link_file_mod['arguments']:
+        if arg['name'] == argname:
+            if 'isBranch' in arg and not arg['isBranch']:
+                return False
+            else:
+                return True
+    return True
+
+
+# Goes through a list and returns true if the given argument branches, recursively checking in 'exclusive' blocks
+def is_branching_in_list(arglist, name):
+    for arg in arglist:
+        if 'exclusive' in arg:
+            if not is_branching_in_list(arg['exclusive'], name):
+                return False
+        if 'name' not in arg:
+            continue
+        if arg['name'] == name:
+            if 'isBranch' in arg and not arg['isBranch']:
+                return False
+            else:
+                return True
+    return True
+
+
+def check_args(module_1, module_2, link_module):
+    # For each argument in the first module
+    if 'arguments' in module_1:
+        if 'arguments' not in module_2:
+            return False
+        for arg in module_1['arguments']:
+            argname = list(arg.keys())[0]
```

Comment:
Ah ic, it wasn't clear that this was checking the config file and the `list()[0]` made it seem like it was a long list. If it's always (assumed) to be only one (key, value) pair, then the following will unpack it:

```python
argname, = arg # or arg.keys() to be visually explicit
match, = filter(lambda x: argname in x, module_2['arguments']) # again, or x.keys() to be visually specific
```

or this should do everything at once:

```python
mod1 = sorted(module_1['arguments'], key=lambda x: list(x))
mod2 = sorted(module_2['arguments'], key=lambda x: list(x))
for arg1, arg2 in zip(mod1, mod2):
    if list(arg1) != list(arg2) or list(arg1.values()) != list(arg2.values()):
        return False
```